### PR TITLE
Robyn one pager end to end flow for tutorial 1

### PR DESCRIPTION
**Pull request by sumalreddy17**
Fix end-to-end one pager generation flow

* Remove redundant parameter passing
* Fix method call in plot_one_pager

- Update type hints and parameters
* Change List[str] to List[PlotType] for plots parameter
* Update ClusteringConfig to ClusteredResult
* Add proper return type hints

- Improve error handling
* Add better logging messages
* Add proper figure return handling

 Add single solution selection option

- Allow passing specific solution_id to plot_one_pager
* Maintain default top pareto behavior
* Update documentation with usage examples"

Testing: Verified one pager generation from Robyn class works end-to-end"

Screenshot 2024-11-22 at 10 16 22 AM

### CODE REVIEW

File: python/src/robyn/robyn.py [hunk 1]
```diff
@@ -398,29 +400,49 @@ def optimize_budget(
             logger.error("Budget optimization failed: %s", str(e))
             raise
 
-    @staticmethod
-    def generate_one_pager(
-        pareto_result: ParetoResult,
-        cluster_result: Optional[ClusteringConfig] = None,
-        mmm_data: Optional[MMMData] = None,
-        plots: Optional[List[str]] = None,
+    def plot_one_pager(self, solution_id: Optional[str] = None) -> None:
```
Comment (sumane81) [inline python/src/robyn/robyn.py:403]: This wrapper method on top of _generate_one_pager doesn't do anything. I would recommend just make generate_one_pager method public. Also, let's plot all plots from this method. No need to keep plotType input parameter here in this interface.

File: python/src/robyn/reporting/onepager_reporting.py
Comment (sumane81) [inline python/src/robyn/reporting/onepager_reporting.py:1]: See some formatter changes. Can you confirm if you are using Black Formatter in VSCode for formatting this?

Comment (sumalreddy17) [reply python/src/robyn/reporting/onepager_reporting.py:1]: yes, confirmed that it is formatted using black formatter.

**Review by sumane81 — APPROVED** [review]
Looks great.

**Pull request merged**